src/downloader.ts

```diff
@@ -4,36 +4,28 @@ import {Readable} from 'stream'
 import unzipper from 'unzipper'
 import {spawn} from 'child_process'
 import {delimiter} from 'path'
+import fetch from 'node-fetch'
 
 const gitForWindowsUsrBinPath = 'C:/Program Files/Git/usr/bin'
 const gitForWindowsMINGW64BinPath = 'C:/Program Files/Git/mingw64/bin'
 
 async function fetchJSONFromURL<T>(url: string): Promise<T> {
-  return new Promise<T>((resolve, reject) => {
-    https
-      .request(url, {}, res => {
-        if (res.statusCode !== 200) {
-          reject(
-            new Error(
-              `Got code ${res.statusCode}, URL: ${url}, message: ${res.statusMessage}`
-            )
+  return new Promise<T>(async (resolve, reject) => {
+    try {
+      const res = await fetch(url)
```

Comment:
Since `fetch()` is _already_ async, we can probably get rid of the `return new Promise<T>(...)` business altogether, and simply `return (await fetch(url)).json() as T`, right?

Reply:
Yes, but then you won't be able to the check for server-side errors (`if (res.status !== 200)`) as `.json()` simply returns the response body without any further checks. Client-side errors will be thrown as exceptions, so those you _will_ see as an error.

Reply:
Right, you need to separate the `await fetch()` from the `.json()`. But still, I think we should get rid of that awful explicit `new Promise()` because it is not needed.

Reply:
You're right. Done in https://github.com/git-for-windows/setup-git-for-windows-sdk/pull/11/commits/1a3b434794c61983c7e4e9156988eb28371b7948 🚀